### PR TITLE
add openai api key support to analyst

### PR DESCRIPTION
**Pull request by briehl**
As it says - this assumes that the AnalystAgent requires an openai api key in order to work (which it does!).
It also assumes that the user will either pass it in as a kwarg or have it loaded as an environment variable (the standard OPENAI_API_KEY one).
If neither of those are the case, it'll throw a KeyError (though maybe this should be a ValueError, or make up a MissingAPIKeyError later).

Tests are still todo here.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -163,6 +163,7 @@ cython_debug/
 # Private files
 
 config.py
+.DS_Store
```
Comment (briehl) [inline .gitignore:166]: Every mac folder seems to get one of these (I think it's used for the os search indexing). These should get ignored.

File: narrative_llm_agent/agents/analyst.py [hunk 2]
```diff
@@ -6,62 +6,138 @@
...
+from pathlib import Path
 
 
 class AnalystInput(BaseModel):
-    input: str = Field(description="query to look up KBase documentation, catalog or tutorials")
+    input: str = Field(
+        description="query to look up KBase documentation, catalog or tutorials"
+    )
+
+DEFAULT_CATALOG_DB_DIR: Path = Path(__file__).parent / "vector_db_app_catalog"
```
Comment (briehl) [inline narrative_llm_agent/agents/analyst.py:19]: Set the default paths here.
This PR also migrates the folders from `tests/agents` to `agents`

File: narrative_llm_agent/agents/analyst.py [hunk 3]
```diff
@@ -6,62 +6,138 @@
...
+    _docs_db_dir: Path
 
-    def __init__(self: "AnalystAgent", token: str, llm: LLM):
+    def __init__(
+        self: "AnalystAgent",
+        token: str,
+        llm: LLM,
+        openai_api_key: str = None,
+        catalog_db_dir: Path | str = None,
+        docs_db_dir: Path | str = None,
```
Comment (briehl) [inline narrative_llm_agent/agents/analyst.py:40]: These are now parameterized as kwargs. Should make it easier for testing with different options, though the defaults still remain.

File: narrative_llm_agent/agents/analyst.py [hunk 4]
```diff
@@ -6,62 +6,138 @@
...
+                f"Database directory {db_path} is not a directory, unable to make Agent."
+            )
+        db_file = db_path / "chroma.sqlite3"
+        if not db_file.exists():
+            raise RuntimeError(
+                f"Database file {db_file} not found, unable to make Agent."
+            )
 
-        @tool("Kbase documentation retrieval tool", args_schema = AnalystInput, return_direct=True)
+    def __setup_openai_api_key(self, openai_api_key: str) -> None:
```
Comment (briehl) [inline narrative_llm_agent/agents/analyst.py:80]: If we have a key as a parameter, great. If not, look for the environment variable. If that doesn't exist either, throw an error.

File: narrative_llm_agent/agents/analyst.py [hunk 5]
```diff
@@ -6,62 +6,138 @@
...
-            tools=[kbase_appCatalog_retrieval_tool, kbase_docs_retrieval_tool]
+            tools=[kbase_appCatalog_retrieval_tool, kbase_docs_retrieval_tool],
+        )
+
+    def _create_doc_chain(self, persist_directory: str | Path):
+        # Embedding functions to use
+        embeddings = OpenAIEmbeddings(openai_api_key=self._openai_key)
+        # Use the persisted database
+        vectordb = Chroma(
+            persist_directory=str(persist_directory), embedding_function=embeddings
```
Comment (briehl) [inline narrative_llm_agent/agents/analyst.py:127]: Pathlib is nice, but still needs to get cast as a str for langchain to handle it. Which is silly as pathlib has been accepted as the main tool for a while...

**Review by pgupta11 — APPROVED** [review]

**Pull request merged**